README.md

```diff
@@ -114,13 +115,39 @@ an API-key and the resource type. To read from a table you should also specify t
 
 ### Writing
 
-To write to a resource you'll need to register a DataFrame that was read from
+There are two ways you can use cdp-spark-datasource to write to CDF: using `insertInto` or using the `save` function. 
+- `insertInto` - will check that all fields are present and in the correct order, and can be more convenient when
+working with Spark SQL tables.
+- `save` - will give you control over how to handle potential collisions with existing data,
+and allows updating a subset of fields in a row. 
+
+#### Writing with `.insertInto()`
+
+To write to a resource using the insert into pattern you'll need to register a DataFrame that was read from
 that resource as a temporary view. You'll need a project where you have write access
 and replace `myApiKey` in the examples below.
 
 Your schema will have to match that of the target exactly. A convenient way to ensure
 this is to copy the schema from the DataFrame you read into with `sourceDf.select(destinationDf.columns.map(col):_*)`, see time series example.
 
+`.insertInto()` will do upsert, as in updating existing rows and inserting new rows, for events, assets and time series.
+It will do insert for raw tables and data points, and throw an error if one or more rows already exist.
+
+#### Writing with `.save()`
+
+Writing with `.save()` is currently supported for assets, events and time series.
+
+You'll need to provide an API-key and the event type you'd like to write to. In addition you can specify
+the desired behaviour when rows in your Dataframe are present in CDF with the `.option("onconflict", value)`.
+
+The valid options for onconflict are
+- `abort` - will try to insert all rows in the Dataframe. An error will be thrown if the resource item already exists and no rows will be written.
+- `update` - will look for all rows in the Dataframe in CDF and try to update them. If one or more rows are missing, none will be updated.
```

Comment:
I suspect some might be updated as well...? Does it throw an error if something isn't found? Due to batching some previous batch might have succeeded already, I'd guess?

Reply:
Yep, fixed.